src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs

```diff
@@ -260,6 +281,113 @@ public static void BuildChain_WithCertificatePolicy_NoMatch()
+                using (var machineRootStore = new X509Store(StoreName.Root, StoreLocation.LocalMachine))
+                {
+                    machineRootStore.Open(OpenFlags.ReadOnly);
+                    bool foundCert = false;
+
+                    foreach (var machineCert in machineRootStore.Certificates)
+                    {
+                        if (machineCert.Equals(microsoftDotComRoot))
+                        {
+                            foundCert = true;
+                        }
+
+                        machineCert.Dispose();
+                    }
+
+                    Assert.True(foundCert, string.Format("Did not find expected certificate with thumbprint '{0}' in the machine root store", microsoftDotComRoot.Thumbprint));
+                }
+
+                // Concievably at this point there could still be something wrong and we still don't chain build correctly - if that's 
+                // the case, then there's likely something wrong with the machine. Validating that happy path is out of scope 
+                // of this particular test - instead, use BuildChain_MicrosoftDotCom_WithSystemRootCerts as a canary 
+
+
+                // Check that microsoft.com's root certificate is NOT installed on in the user cert store as a sanity step
+                // We won't try to install the microsoft.com root cert into the user root store if it's already there
+                using (var userRootStore = new X509Store(StoreName.Root, StoreLocation.CurrentUser))
+                {
+                    userRootStore.Open(OpenFlags.ReadOnly);
+
+                    foreach (var userCert in userRootStore.Certificates)
+                    {
+                        bool foundCert = false;
+                        if (userCert.Equals(microsoftDotComRoot))
+                        {
+                            foundCert = true;
+                        }
+
+                        userCert.Dispose();
+
+                        if (foundCert)
+                        {
+                            shouldInstallCertToUserStore = false;
+                        }
+                    }
+                }
+
+                using (var userRootStore = new X509Store(StoreName.Root, StoreLocation.CurrentUser))
+                using (var chainHolder = new ChainHolder())
+                {
+                    try
+                    {
+                        if (shouldInstallCertToUserStore)
+                        {
+                            userRootStore.Open(OpenFlags.ReadWrite);
+                            userRootStore.Add(microsoftDotComRoot); // throws CryptographicException
+                            installedCertToUserStore = true;
+                        }
+
+                        X509Chain chainValidator = chainHolder.Chain;
+                        chainValidator.ChainPolicy.VerificationTime = new DateTime(2015, 10, 15, 12, 01, 01, DateTimeKind.Local);
```

Comment:
I lifted this date/time from 

```
                // Halfway between microsoftDotCom's NotBefore and NotAfter
                // This isn't a boundary condition test.
                chain.ChainPolicy.VerificationTime = new DateTime(2015, 10, 15, 12, 01, 01, DateTimeKind.Local);
                chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
```

I figure that since this was successfully used before, I'll just play copycat :)